From d9b782307205d313b78daa901f36c72d3a86fd66 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 06 Feb 2023 11:45:43 +0200
Subject: [PATCH] [SW-99614] gaudi2: security: remove long rr code

Neither Ppboot, nor Preboot is using long RRs, so the long RR code patch
is actually a dead code which cannot be removed by garbage collection as
it is a part of a function that is in use and thus is just a dead code.
Remove long RR support and thus remove the dead code path.

Change-Id: Id40747730be04541ebb0b7625cb32c4c4a5377f1
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index e7529a2..fcd8472 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -278,8 +278,7 @@
 	hl_secure_update_hit_reg(rr_idx, inst_offset, &rr_reg->hit, false);
 }
 
-#define MAX_LBW_SHORT_RR	14
-
+/* hl_secure_lbw_rr_gen() function deals only with short RRs rr_idx < 14*/
 static void hl_secure_lbw_rr_gen(int rr_idx, u64 min, u64 max, int dist,
 		int inst_num, enum hl_sec_engine_type type)
 {
@@ -290,17 +289,11 @@
 	rr_reg.data_min = (min & ~CFG_BASE);
 	rr_reg.data_max = (max & ~CFG_BASE);
 
-	if (rr_idx < MAX_LBW_SHORT_RR) {
-		rr_reg.rr_idx_calc = rr_idx;
-		rr_reg.data_min >>= HL_SEC_SHIFT_4K_POS;
-		rr_reg.data_max >>= HL_SEC_SHIFT_4K_POS;
-		rr_reg.reg_min = cfg->min_short_breg;
-		rr_reg.reg_max = cfg->max_short_breg;
-	} else {
-		rr_reg.rr_idx_calc = rr_idx - MAX_LBW_SHORT_RR;
-		rr_reg.reg_min = cfg->min_full_breg;
-		rr_reg.reg_max = cfg->max_full_breg;
-	}
+	rr_reg.rr_idx_calc = rr_idx;
+	rr_reg.data_min >>= HL_SEC_SHIFT_4K_POS;
+	rr_reg.data_max >>= HL_SEC_SHIFT_4K_POS;
+	rr_reg.reg_min = cfg->min_short_breg;
+	rr_reg.reg_max = cfg->max_short_breg;
 
 	rr_reg.reg_min += dist;
 	rr_reg.reg_max += dist;
